modules/appeals_api/app/controllers/concerns/appeals_api/caseflow_request.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module CaseflowRequest
+    extend ActiveSupport::Concern
+
+    # Redefine this in the including controller
+    def get_caseflow_response
+      raise 'Unable to get response from caseflow'
+    end
+
```

Comment:
Here's the concern that provides the shared caseflow logic. As I was rewriting the Appeals Status controller, I noticed there were aspects of the code that communicated with Caseflow which were duplicated across a few APIs:


Handling of `BackendServiceException`s used to look like this:
  - [In Appealable Issues](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/app/controllers/appeals_api/appealable_issues/v0/appealable_issues_controller.rb#L60-L68) (supporting methods are defined in its [parent controller, here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/contestable_issues_controller.rb#L57-L64))
  - [In Legacy Appeals](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/legacy_appeals_controller.rb#L60-L64)


**4XX errors:**

In APIs listed above, if `caseflow_returned_a_4xx?` is true, the error body is returned as-is after removing some extra unnecessary information. 

[In Appeals Status](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb#L12-L23), on the other hand, there doesn't seem to be any existing special error handling - Caseflow errors are returned without reformatting. This doesn't seem correct, and I've previously [called it out in our docs specs](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/appeals_api/spec/docs/appeals_status/v1_spec.rb#L102) as something to fix.

**5XX errors:**

In previous code, if the error is >= 500, it is re-raised. The [Lighthouse standards](https://department-of-veterans-affairs.github.io/lighthouse-api-standards/errors/#choosing-an-error-code) instead encourage us to use a 502 when we get a 500 from upstream (instead of letting it cause a 500 in our own API).

**Updates:**

In light of all this, I added this error handling below to do the following:
* Handle any `BackendServiceException` thrown from a Caseflow request so that:
  * 4XX errors are still handled as before - modifying the body so that we don't expose more information than necessary, and then reusing that body and status to pass along to the user
  * 5XX errors are caught and returned as generic 502 errors instead of being reraised
* Log the same relevant information as before (or more in some cases)
* Attempt to make sure any errors that are returned match our `errorModel` schema - this is done by trying to raise predefined errors from `Common::Exceptions` wherever we can.

Reply:
> letting it cause a 500 in our own API
Nice catch!